3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +23,59 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw 'network error';
+  } else {
+    return response.json();
+  }
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchAndPopulatePokemons() {
+  if (document.body.childElementCount >= 4) {
+    return;
+  } else {
+    const selectElement = document.body.appendChild(
```

Comment:
This is a very fragile test. If another element is added to the body this test will fail. If you just want to test whether a select element was already created, you could just test whether that element is already in the DOM. And for reasons mentioned earlier, you don't need `else` after `return`.

```suggestion
  if (document.querySelector('select')) {
    return;
  }
  const selectElement = document.body.appendChild(
```